File: apps/dashboard/types/publisher-app.ts
export type Platform = "ios" | "android" | "web";

export interface PublisherApp {
  id: string;
  org_id: string;
  name: string;
  platform: Platform;
  bundle_id: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreatePublisherAppRequest {
  name: string;
  platform: Platform;
  bundle_id: string;
}

export interface UpdatePublisherAppRequest {
  name?: string;
  is_active?: boolean;
}

export interface PublisherAppListResponse {
  items: PublisherApp[];
  total: number;
  limit: number;
  offset: number;
}

File: services/api-dashboard/internal/model/publisher_app.go
package model

import (
	"time"

	"github.com/google/uuid"
)

// PublisherApp represents a mobile or web application registered by a publisher organisation.
type PublisherApp struct {
	ID        uuid.UUID `json:"id"`
	OrgID     uuid.UUID `json:"org_id"`
	Name      string    `json:"name"`
	Platform  string    `json:"platform"`
	BundleID  string    `json:"bundle_id"`
	IsActive  bool      `json:"is_active"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}
